src/other/index.jl

```diff
@@ -96,6 +96,14 @@ function Base.delete!(x::Index, idx::Integer)
     return x
 end
 
+"""
+Returns returns the indices of the columns in x that are not in y. 
+"""
+function diff_indices(x::Index, y::Index)
```

Comment:
Sorry, when I suggested having a separate function for this, I was thinking about a `MetaData`-aware function which would avoid calling `setdiff` if the meta-data is empty. But it can't live in index.jl, which is only about `Index`. Since the function is just `x[setdiff(names(x), names(y))]`, it's not very useful. Maybe just move all that stuff into `append!` and pass it the two `Index` objects. That way it can be a no-op when meta-data is empty. I guess it depends on what can be shared with the `join` code.